src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/ManagedIndexSweeper.kt

```diff
@@ -0,0 +1,370 @@
+            if (!created) {
+                logger.debug("Failed to create $INDEX_STATE_MANAGEMENT_INDEX")
+                return
+            }
+        }
+
+        doBulkRequests(createManagedIndexRequests + deleteManagedIndexRequests + updateManagedIndexRequests)
+    }
+
+    /**
+     * Sweeps the [INDEX_STATE_MANAGEMENT_INDEX] for ManagedIndices.
+     *
+     * Sweeps the [INDEX_STATE_MANAGEMENT_INDEX] for ManagedIndices and only fetches the index, index_uuid,
+     * policy_name, and change_policy fields to convert into a [SweptManagedIndex].
+     *
+     * @return map of IndexUuid to [SweptManagedIndex].
+     */
+    private suspend fun sweepManagedIndexJobs(): Map<String, SweptManagedIndex> {
+        if (!ismIndices.indexStateManagementIndexExists()) return mapOf()
+
+        val boolQueryBuilder = BoolQueryBuilder().filter(QueryBuilders.existsQuery(ManagedIndex.MANAGED_INDEX_TYPE))
+        val managedIndexSearchRequest = SearchRequest()
+                .indices(INDEX_STATE_MANAGEMENT_INDEX)
+                .source(SearchSourceBuilder.searchSource()
+                        // TODO: Get all ManagedIndices at once or split into searchAfter queries?
+                        .size(MAX_HITS)
+                        .fetchSource(
+                                arrayOf(
+                                        "${ManagedIndex.MANAGED_INDEX_TYPE}.${ManagedIndex.INDEX_FIELD}",
+                                        "${ManagedIndex.MANAGED_INDEX_TYPE}.${ManagedIndex.INDEX_UUID_FIELD}",
+                                        "${ManagedIndex.MANAGED_INDEX_TYPE}.${ManagedIndex.POLICY_NAME_FIELD}",
+                                        "${ManagedIndex.MANAGED_INDEX_TYPE}.${ManagedIndex.CHANGE_POLICY_FIELD}"
+                                ),
+                                emptyArray()
+                        )
+                        .query(boolQueryBuilder))
+
+        val response: SearchResponse = client.suspendUntil { search(managedIndexSearchRequest, it) }
+        val sweptManagedIndices = mutableMapOf<String, SweptManagedIndex>()
+
+        response.hits.forEach {
+            val sweptManagedIndex =
+                    SweptManagedIndex.parseWithType(SweptManagedIndex.parser(it.sourceRef), it.seqNo, it.primaryTerm)
+            sweptManagedIndices[sweptManagedIndex.uuid] = sweptManagedIndex
+        }
+
+        return sweptManagedIndices
+    }
+
+    /**
+     * Sweeps the cluster state for ManagedIndices.
+     *
+     * Sweeps the cluster state for ManagedIndices by checking for the [POLICY_NAME] in the index settings.
+     * If the [POLICY_NAME] is null or blank it's treated as not existing.
+     *
+     * @return map of IndexUuid to [ClusterStateManagedIndex].
+     */
+    private fun sweepClusterState(): Map<String, ClusterStateManagedIndex> {
+        val clusterStateManagedIndices = mutableMapOf<String, ClusterStateManagedIndex>()
+        clusterService.state().metaData().indices().values().forEach {
```

Comment:
shall we filter out the closed indices?

Reply:
We support a "close" and "open" actions which means we need to keep track of the "closed" indices that still have policy names in case they will be opened again.